includes/class-coblocks-block-assets.php

```diff
@@ -479,6 +490,21 @@ public function has_coblocks_block( WP_Post $post_object ) {
 
 	}
 
+	/**
+	 * Determine if the given post content contains any v1 Masonry block.
+	 *
+	 * @access public
+	 * @since  2.22.0
+	 *
+	 * @return boolean True when post content contains a v1 Masonry block.
+	 */
+	public function has_masonry_v1_block() {
+		$v1_regex = '/<!-- wp:coblocks\/gallery-masonry.*|\n*(coblocks-gallery--item).*|\n*<!-- \/wp:coblocks\/gallery-masonry -->/m';
```

Comment:
You shouldn't need the multi-line flag (`/m`) here because you're not using the beginning string (`^`) and end of string (`$`) modifiers. Just the slash without any flags would work.

Reply:
Posting here for reference a possible new regex so we can simplify the function: https://regex101.com/r/xESuJM/1

```
/<!--\s+?wp:coblocks\/gallery-masonry[\s\S]*?coblocks-gallery--item/
```

Reply:
This would also work:
```
/<!--\s+?wp:coblocks\/gallery-masonry.*?coblocks-gallery--item/s
```